File: Backend/UFVQuest/UFVQuestAPI/ufvquest_utils.py
from math import radians, cos, sin, asin, sqrt


#Calculate distance in meters between the Lat and Lon passed and the Centro de Vivencia (-20.761717, -42.869958)
def distance_to_centro_de_vivencia(lat, lon):
	lon1, lat1, lon2, lat2 = map(radians, [lon1, lat1, -20.761717, -42.869958])
    # haversine formula 
    dlon = lon2 - lon1 
    dlat = lat2 - lat1 
    a = sin(dlat/2)**2 + cos(lat1) * cos(lat2) * sin(dlon/2)**2
    c = 2 * asin(sqrt(a)) 
    km = 6367 * c
    return (km * 1000)

File: Web/js/ufv-quest-utils.js
var API_URL = "http://107.170.81.127:8777/UFVQuestAPI";
var USER;

function getFormObj(form) {
    var formObj = {};
    var inputs = $(form).serializeArray();
    $.each(inputs, function (i, input) {
        formObj[input.name] = input.value;
    });
    return formObj;
}

Number.prototype.toRad = function() {
   return this * Math.PI / 180;
}

function getDistanceBetween(latlng1, latlng2) {
	
	var lat2 = latlng2.lat(); 
	var lon2 = latlng2.lng(); 
	var lat1 = latlng1.lat(); 
	var lon1 = latlng1.lng(); 

	var R = 6371; // km 
	var x1 = lat2-lat1;
	var dLat = x1.toRad();  
	var x2 = lon2-lon1;
	var dLon = x2.toRad();  
	var a = Math.sin(dLat/2) * Math.sin(dLat/2) + 
					Math.cos(lat1.toRad()) * Math.cos(lat2.toRad()) * 
					Math.sin(dLon/2) * Math.sin(dLon/2);  
	var c = 2 * Math.atan2(Math.sqrt(a), Math.sqrt(1-a)); 
	var d = R * c * 1000; 
	
	return d;
}